src/Model/ModelManager.php

```diff
@@ -238,6 +240,22 @@ public function reverseTransform(object $object, array $array = []): void
         }
     }
 
+    public function getRealClass(object $object): string
+    {
+        $class = \get_class($object);
+
+        $dm = $this->registry->getManagerForClass($class);
+        if (null === $dm) {
+            return $class;
+        }
+
+        try {
+            return $dm->getClassMetadata($class)->getName();
```

Comment:
In this same class we have a getMetadata method that throws exceptions when the class is not managed by mongo or has not metadata, cant we use that, and catch those exception instead? feels like duplicated code now that I look again.

Reply:
We cannot for static analysis reason.
```
/**
     * @param class-string<T> $class
     *
     * @return MongoDBClassMetadata<T>
     */
    private function getMetadata(string $class): MongoDBClassMetadata
    {
        return $this->getDocumentManager($class)->getClassMetadata($class);
    }
```
only accepts class-string<T>. But here we are passing any classes.

Reply:
But maybe changing to
```
/**
    * @template S
     * @param class-string<S> $class
     *
     * @return MongoDBClassMetadata<S>
     */
    private function getMetadata(string $class): MongoDBClassMetadata
    {
        return $this->getDocumentManager($class)->getClassMetadata($class);
    }
```
could do the trick...